### PR TITLE
Fix https://supabase.sentry.io/issues/6037628167/

### PR DESCRIPTION
**Pull request by joshenlim**
- Reproduced by clicking "Add new row" on a table that has 0 columns
- If a table has 0 columns, postgrest returns the `columns` property as null instead of an empty array
- This caused generateRowFields to error out, also due to the unnecessary `!` parameter
- Hide Add new row action if table has 0 columns

### CODE REVIEW

**Comment by MildTomato** [comment]
confirmed fixed on preview

File: apps/studio/components/interfaces/TableGridEditor/SidePanelEditor/ColumnEditor/ColumnForeignKey.tsx [hunk 1]
```diff
@@ -116,13 +116,20 @@ const ColumnForeignKey = ({
...
             name: table.name,
-            columns:
-              column.isNewColumn && column.name
-                ? (table.columns as any[]).concat(column)
-                : (table.columns as any[]).map((c) => {
-                    if (c.id === column.id) return { ...c, name: column.name }
-                    else return c
-                  }),
+            columns: (column.isNewColumn && column.name
+              ? (table.columns ?? []).concat(column as any)
```
Comment (MildTomato) [inline apps/studio/components/interfaces/TableGridEditor/SidePanelEditor/ColumnEditor/ColumnForeignKey.tsx:120]: `any` ?

Comment (MildTomato) [reply apps/studio/components/interfaces/TableGridEditor/SidePanelEditor/ColumnEditor/ColumnForeignKey.tsx:120]: it was `any` before so not a blocker.

Comment (joshenlim) [reply apps/studio/components/interfaces/TableGridEditor/SidePanelEditor/ColumnEditor/ColumnForeignKey.tsx:120]: hmm actually lemme just clean this up too

Comment (joshenlim) [reply apps/studio/components/interfaces/TableGridEditor/SidePanelEditor/ColumnEditor/ColumnForeignKey.tsx]: okay pushed a refactor up!

**Review by alaister — APPROVED** [review]
LGTM!

- Tested on preview

**Pull request merged**